src/plugins/query_enhancements/server/routes/query_assist/ppl/time_parser_utils.ts

```diff
@@ -0,0 +1,211 @@
+      for (const mappingIndexName in mappingResponse) {
+        if (!Object.prototype.hasOwnProperty.call(mappingResponse, mappingIndexName)) continue;
+        const { mappings } = mappingResponse[mappingIndexName];
+        for (const fieldKey in mappings) {
+          if (!Object.prototype.hasOwnProperty.call(mappings, fieldKey)) continue;
+          const fieldDef = mappings[fieldKey];
+          const innerMappings = fieldDef.mapping ?? {};
+          for (const innerFieldName in innerMappings) {
+            if (!Object.prototype.hasOwnProperty.call(innerMappings, innerFieldName)) continue;
+            const def = innerMappings[innerFieldName];
+            if (def.type === 'date' || def.type === 'date_nanos') {
+              dateFields.add(fieldDef.full_name);
+              // Initialize cluster for this date field
+              if (!clusterMap[fieldDef.full_name]) {
+                clusterMap[fieldDef.full_name] = new Set([fieldDef.full_name]);
+              }
+            }
+          }
+        }
+      }
+      // Step 2: Find all fields of type 'alias' that point to a date field
+      for (const mappingIndexName in mappingResponse) {
+        if (!Object.prototype.hasOwnProperty.call(mappingResponse, mappingIndexName)) continue;
+        const { mappings } = mappingResponse[mappingIndexName];
+        for (const fieldKey in mappings) {
+          if (!Object.prototype.hasOwnProperty.call(mappings, fieldKey)) continue;
+          const fieldDef = mappings[fieldKey];
+          const innerMappings = fieldDef.mapping ?? {};
+          for (const innerFieldName in innerMappings) {
+            if (!Object.prototype.hasOwnProperty.call(innerMappings, innerFieldName)) continue;
+            const def = innerMappings[innerFieldName];
+            if (def.type === 'alias' && typeof def.path === 'string') {
+              const targetField = def.path;
+              const aliasField = fieldDef.full_name;
+              if (dateFields.has(targetField)) {
+                if (!clusterMap[targetField]) {
+                  clusterMap[targetField] = new Set([targetField]);
+                }
+                clusterMap[targetField].add(aliasField);
+              }
+            }
+          }
+        }
+      }
+      // Convert clusters from Set<string> to string[][]
+      return Object.values(clusterMap).map((set) => Array.from(set));
+    }
+
+    return parseTimestampAliasClusters(response.body);
+  } catch (error) {
+    logger.error(`Error retrieving timestamp field clusters for index ${indexName}: ${error}`);
+    throw error;
+  }
+}
+
+/**
+ * Get all time fields (flattened) in the index except the selected field
+ * @param indexName - Index name
+ * @param selectedTimeField - The selected time field
+ * @param client - OpenSearch client
+ * @param logger - Logger
+ * @param getTimestampFieldClustersFn - Optional function to get timestamp field clusters (for testing)
```

Comment:
We can use jest.mock to hijack the function, not a good practice to add an optional parameters that is only used in UT.